Source/RpmAvatarCreator/Private/UI/RpmAvatarEditorUI.cpp

```diff
@@ -200,3 +257,72 @@ void URpmAvatarEditorUI::SetColorSelectedPin(ERpmPartnerAssetColor AssetColor, i
 		ColorButton->SetSelected(IsSelected);
 	}
 }
+
+void URpmAvatarEditorUI::DownloadAssetIcons(const UWrapBox* AssetContainer)
+{
+	for (UWidget* Widget : AssetContainer->GetAllChildren())
+	{
+		URpmAssetButtonUI* AssetButton = Cast<URpmAssetButtonUI>(Widget);
+		const FString& IconUrl = AssetButton->Asset.IconUrl;
+		if (!IconUrl.IsEmpty())
+		{
+			AssetImageMap.Add(IconUrl, AssetButton);
+			AvatarCreatorApi->DownloadImage(IconUrl, 64, AssetIconDownloadCompleted);
```

Comment:
similar to the other comment about the hardcoded 64. Perhaps the actual badge icon doesn't need to be adjustable but the asset icons probably should be. 
In general I don't particularly like the hard coded sizes, especially if they are done in multiple places, it's harder to maintain.